### PR TITLE
Add option to generate schema example

### PR DESCRIPTION
**Pull request by ottomata**
If shouldGenerateExample is true, an example will be generated during
schema materialization.

Examples already present in the schema will be preserved. E.g. If the
schema already has examples, and shouldGenerateExample is true,
no new example will be generated.

Bug: T270134

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -295,10 +295,11 @@ configPaths: ['./.jsonschema-tools.yaml']
...
 
-# If true, repository tests for schemas will fail if the schema doesn't have
-# examples.  If it does have examples, they will be also tested for
-# validation against the schema.
-requireExamples: false
+# If true an example will be generated during schema materialization.
+# Examples already present in the schema will be preserved. E.g. If the
+# schema already has examples, and shouldGenerateExample is true,
+# no new example will be generated.
+shouldGenerateExample: false
```
Comment (mdholloway) [inline README.md:302]: This kind of gets into software design philosophy (configurability vs. just doing the Right Thing), but should this really be mutually exclusive with `requireExamples`?  I can imagine someone wanting both to require examples and to require schema implementers to create them by hand.

(Also, where is `requireExamples` enforced? The only place I can find it is here in the README file.)

File: lib/jsonschema-tools.js [hunk 2]
```diff
@@ -108,8 +109,39 @@ const defaultOptions = {
...
+     * no new example will be generated.
+     */
+    shouldGenerateExample: false,
 };
 
+/**
+ * options for json-schema-faker when generating schema examples.
+ */
+const jsonSchemaFakerOptions = {
+    // If useExampleValues is true and an optional field is missing
```
Comment (marcelrf) [inline lib/jsonschema-tools.js]: typo: useExampleValues should be singular, no?

File: README.md (hunk 1 above)
Comment (ottomata) [reply README.md:302]: I removed it in a previous patch, but forgot to remove it from the README at that time.

This `shouldGenerateExamples` flag covers the same use case that `requireExamples` does for us, so I removed `requireExamples` as an option.  Setting `shouldGenerateExample` will ensure that every schema has an example.  I could see someone wanting to require examples without generating them, but let's wait until that is a needed use case.

Also, `requireExamples` was only enforced in repo tests, not by any materialization logic.  If someone wasn't using the repo tests, `requireExamples` wouldn't have any effect.  I'd actually like to refactor some of the way we apply these tests.  Right now it is difficult, if not impossible, to choose which tests you'd like to apply.  I've got three categories of repo tests, and it is possible to hack around which ones you run, but for the individual test cases, there's no way to selectively choose them.  Ideally the tests would work more like a linter, with easy configuration of what lint rules to enforce.  By not including `requireExamples` as a regular config...we won't have to figure out how to move it into a future lint rules config if we ever are able to create them.

**Review by marcelrf — COMMENTED** [review]
Looks good to me! Left a typo comment.

File: lib/jsonschema-tools.js (hunk 2 above)
Comment (mdholloway) [inline lib/jsonschema-tools.js]: typo, should be `useExamplesValue`

File: lib/jsonschema-tools.js [hunk 3]
```diff
@@ -108,8 +109,39 @@ const defaultOptions = {
...
+    // in an existent schema example,
+    // json-schema-faker will not set that field to a fake value,
+    // even if alwaysFakeOptionals is true.  The best thing to do is
+    // is to not useExamplesValues.
```
Comment (mdholloway) [inline lib/jsonschema-tools.js]: ditto

**Comment by mdholloway** [comment]
Seems to work well.  Aside from the typos, LGTM 👍

**Pull request merged**